pkg/shared/findingkey/malware.go

```diff
@@ -19,22 +19,25 @@ import (
 	"fmt"
 
 	"github.com/openclarity/vmclarity/api/models"
+	"github.com/openclarity/vmclarity/pkg/shared/utils"
 )
 
 type MalwareKey struct {
 	MalwareName string
 	MalwareType string
 	Path        string
+	RuleName    string
 }
 
 func (k MalwareKey) String() string {
-	return fmt.Sprintf("%s.%s.%s", k.MalwareName, k.MalwareType, k.Path)
+	return fmt.Sprintf("%s.%s.%s.%s", k.MalwareName, k.MalwareType, k.Path, k.MalwareName)
 }
 
 func GenerateMalwareKey(info models.MalwareFindingInfo) MalwareKey {
 	return MalwareKey{
-		MalwareName: *info.MalwareName,
-		MalwareType: *info.MalwareType,
-		Path:        *info.Path,
+		MalwareName: utils.StringPointerValOrEmpty(info.MalwareName),
+		MalwareType: utils.StringPointerValOrEmpty(info.MalwareType),
+		Path:        utils.StringPointerValOrEmpty(info.Path),
+		RuleName:    utils.StringPointerValOrEmpty(info.RuleName),
```

Comment:
You'll need to add findingInfo/ruleName back to the $select in the function in assetscanprocessor otherwise this will always be empty.